src/core/Footer/Footer.tsx

```diff
@@ -35,9 +40,15 @@ export type FooterElement = {
    * URL of the footer element.
    */
   url?: string;
+  /**
+   * Key of the footer element.
+   */
+  // eslint-disable-next-line @typescript-eslint/ban-types -- This allows custom strings and keeps intellisense. See https://github.com/Microsoft/TypeScript/issues/29729
+  key?: keyof TitleTranslations | (string & {});
```

Comment:
Can avoid eslint error by using `Record<never, never>`
```suggestion
  // This allows custom strings and keeps intellisense. See https://github.com/Microsoft/TypeScript/issues/29729
  key?: keyof TitleTranslations | (string & Record<never, never>);
```

Reply:
Fixed